src/allocator.rs

```diff
@@ -945,3 +957,59 @@ fn test_roundtrip(#[case] test_value: &str, #[case] make: MakeFun, #[case] check
     let node = make(&mut a, &value);
     check(&mut a, node, &value);
 }
+
+#[cfg(test)]
+#[rstest]
+#[case(&[], 0)]
+#[case(&[1], 1)]
+#[case(&[1,2], 2)]
+#[case(&[1,2,3,4,5,6,7,8,9], 9)]
+#[case(&[1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18], 18)]
+fn test_atom_len(#[case] buf: &[u8], #[case] expected: usize) {
+    let mut a = Allocator::new();
+    let atom = a.new_atom(buf).unwrap();
+    assert_eq!(a.atom_len(atom), expected);
+}
+
+#[cfg(test)]
+#[rstest]
+#[case(0.into(), 0)]
+#[case(42.into(), 1)]
+#[case(127.into(), 1)]
+#[case(1337.into(), 2)]
+#[case(0x7fffff.into(), 3)]
+#[case(0xffffff.into(), 4)]
+#[case((-1).into(), 1)]
+#[case((-128).into(), 1)]
+fn test_atom_len_number(#[case] value: Number, #[case] expected: usize) {
+    let mut a = Allocator::new();
+    let atom = a.new_number(value).unwrap();
+    assert_eq!(a.atom_len(atom), expected);
+}
+
+#[cfg(test)]
+#[rstest]
+#[case("97f1d3a73197d7942695638c4fa9ac0fc3688c4f9774b905a14e3a3f171bac586c55e83ff97a1aeffb3af00adb22c6bb", 48)]
+#[case("a572cbea904d67468808c8eb50a9450c9721db309128012543902d0ac358a62ae28f75bb8f1c7c42c39a8c5529bf0f4e", 48)]
+fn test_atom_len_g1(#[case] buffer_hex: &str, #[case] expected: usize) {
+    let mut a = Allocator::new();
+    let buffer = &hex::decode(buffer_hex).unwrap();
+    let g1 =
+        G1Projective::from(G1Affine::from_compressed(&buffer[..].try_into().unwrap()).unwrap());
+    let atom = a.new_g1(g1).unwrap();
+    assert_eq!(a.atom_len(atom), expected);
+}
+
+#[cfg(test)]
+#[rstest]
+#[case("93e02b6052719f607dacd3a088274f65596bd0d09920b61ab5da61bbdc7f5049334cf11213945d57e5ac7d055d042b7e024aa2b2f08f0a91260805272dc51051c6e47ad4fa403b02b4510b647ae3d1770bac0326a805bbefd48056c8c121bdb8", 96)]
```

Comment:
This might work (and `cargo fmt` will clean it up better):

```suggestion
#[case(
"93e02b6052719f607dacd3a088274f65596bd0d09920b61ab5da61bbdc7f5049"
"334cf11213945d57e5ac7d055d042b7e024aa2b2f08f0a91260805272dc51051"
"c6e47ad4fa403b02b4510b647ae3d1770bac0326a805bbefd48056c8c121bdb8",
96)]
```

Works in python. Don't know offhand in rust though.